### PR TITLE
chore: add duo verification error mock and i18n

### PR DESCRIPTION
**Pull request by TianGan-okta**
## Description:
1. add duo verification error mock
2. add key/value to login bundle
3. add pseudo code

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [x] Added unit tests?

- [x] Added e2e tests

- [x] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:
image

### Reviewers:
@nikhilvenkatraman-okta @jmelberg-okta 

### Issue:

- [OKTA-387804](https://oktainc.atlassian.net/browse/OKTA-387804)

### CODE REVIEW

File: packages/@okta/i18n/src/properties/login.properties [hunk 1]
```diff
@@ -878,6 +878,8 @@ oie.duo.authenticator.description = Verify your identity using Duo Security.
 oie.duo.enroll.title = Set up Duo Security
 oie.duo.verify.title = Verify with Duo Security
 oie.duo.iFrameError = Error loading Duo. Try again or contact your admin for assistance.
+oie.authenticator.duo.method.duo.verification_timeout = We were unable to verify with Duo. Try again.
```
Comment (nikhilvenkatraman-okta) [inline packages/@okta/i18n/src/properties/login.properties:881]: Curious if we know why the backend sending different i18n keys for the same error message ?

**Comment by nikhilvenkatraman-okta** [comment]
Does the content of the duo iframe always render in english ?

File: packages/@okta/i18n/src/properties/login.properties (hunk 1 above)
Comment (TianGan-okta) [reply packages/@okta/i18n/src/properties/login.properties:881]: I think that was running by @ManuMalhotra-okta and @vincentvoong-okta. To me, I still think we can provide different message on backend, but map to same general message on UI as webauthn. I can bring this up to the team tomorrow

**Comment by TianGan-okta** [comment]
> Does the content of the duo iframe always render in english ?

@nikhilvenkatraman-okta yes, created a ticket for it: https://oktainc.atlassian.net/browse/OKTA-392641

File: src/v2/ion/i18nTransformer.js [hunk 2]
```diff
@@ -137,6 +137,10 @@ const I18N_OVERRIDE_MAPPINGS = {
 
   // Remap authn API errors to OIE
   'api.authn.poll.error.push_rejected': 'oktaverify.rejected',
+
+  // Remap duo API errors to OIE
+  'oie.authenticator.duo.method.duo.verification_timeout' : 'oie.authenticator.duo.error',
+  'oie.authenticator.duo.method.duo.verification_failed' : 'oie.authenticator.duo.error'
```
Comment (jmelberg-okta) [inline src/v2/ion/i18nTransformer.js:143]: 👏🏼 This should be the pattern we try to abide to moving forward. Not only do we have a nice `i18n` key to reference, but customizations can be applied to only a single key!

Comment (jmelberg-okta) [inline src/v2/ion/i18nTransformer.js:143]: style-nit: 
```suggestion
  'oie.authenticator.duo.method.duo.verification_timeout': 'oie.authenticator.duo.error',
  'oie.authenticator.duo.method.duo.verification_failed': 'oie.authenticator.duo.error'
```

File: test/unit/spec/v2/ion/i18nTransformer_spec.js [hunk 3]
```diff
@@ -1869,6 +1887,7 @@ describe('v2/ion/i18nTransformer', function() {
       ],
     ])('should render correct error message by get key from API: %s', (key, error, expectedMessage) => {
       Bundles.login = originalLoginBundle;
+      expect(Bundles.login[key]).toBeUndefined();
```
Comment (jmelberg-okta) [inline test/unit/spec/v2/ion/i18nTransformer_spec.js:1890]: 👍🏼

File: playground/mocks/data/idp/idx/error-authenticator-duo-verification-timeout.json [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        ],
+        "accepts": "application/json; okta-version=1.0.0"
+      }
+    ]
+  },
+  "messages": {
+    "type": "array",
+    "value": [
+      {
+        "message": "We did not hear back from Duo.",
```
Comment (jmelberg-okta) [inline playground/mocks/data/idp/idx/error-authenticator-duo-verification-timeout.json:52]: Is this a real error message from Duo or our API? We should consider changing this to something a bit more formal and with a possible resolution.

**Review by jmelberg-okta — APPROVED** [review]
Great work! 🚀

**Review by nikhilvenkatraman-okta — APPROVED** [review]

File: playground/mocks/data/idp/idx/error-authenticator-duo-verification-timeout.json (hunk 4 above)
Comment (ManuMalhotra-okta) [reply playground/mocks/data/idp/idx/error-authenticator-duo-verification-timeout.json:52]: The server side is sending this message.
`
We were unable to verify with Duo. Try again.
`

https://github.com/okta/okta-core/blob/5750e25bfed06127c8dc6993d0f806777fdf927b/resources/src/main/resources/api-messages.properties#L1004

**Pull request closed without merging**